// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package pantheon;

service PantheonInterface {
  rpc ReceiveParams(Info) returns (CryptoParams) {}
  rpc SendKeys(CryptoKeys) returns (Info) {}
  rpc SendOneCiphertext(OneCiphertext) returns (Info) {}
  rpc Query(QueryStream) returns (ResponseStream) {}
}

message Info { string info = 1; }

message CryptoParams { bytes parms_ss = 1; }

message CryptoKeys { bytes keys_ss = 1; }

message OneCiphertext { bytes one_ct_ss = 1; }

message QueryStream { bytes qss = 1; }

message ResponseStream { bytes ss = 1; }
